strapi/strapi
# [SSO lock feature] Add locking to LocalStrategy

### What does it do?

When a role is SSO locked:

- Password cannot be changed
- Forgot password is not sent
- LocalStrategy throws an error

### Why is it needed?

Prevent certain roles from logging in with Local Auth instead of SSO.

### How to test it?

Create a user with a role like 'editor'

Test the locking works with EE:
- In the SSO settings page, lock the 'editor' role to SSO-only
- SSO logins for the user should work
- Local (username+password) login should be blocked
- Forget password for the user should no longer send an email
- Reset password for the user should not work

Test that functionality still works with CE (follow previous steps, then remove license):
- Forget password should still send email even if it was locked while EE was enabled
- reset password should still work even if it was locked while EE was enabled

Test that unlocked functionality still works with EE:
- Unlock the 'editor' role
- Forget password should now send email
- reset password should now work
